crates/pera-core/src/dwallet_mpc/mpc_instance.rs

```diff
@@ -112,6 +112,33 @@ impl DWalletMPCInstance {
         Ok(())
     }
 
+    fn restart(&mut self) {
+        self.pending_messages.clear();
+        self.status = MPCSessionStatus::FirstExecution;
+        self.party = match &self.session_info.mpc_round {
+            MPCRound::DKGFirst => {
```

Comment:
I'm confused: we're setting the `status` to `FirstExecution`, but then select the `party` based on the `session_info.mpc_round`? Can you explain what is supposed to happen here?

Reply:
We need to restart that instance, which require setting it's status to first execution, so it will be executed again in the next round without any pending messages, and also re-create the first party, as it's being consumed by the advance call